### ISSUE TITLE
Unhandled Exception: System.AccessViolationException: (Windows)

### ISSUE THREAD

**Issue by wsnoble** [crux-toolkit/crux-toolkit#380]
A user reports that the pipeline command throws an error:

Unhandled Exception: System.AccessViolationException: Attempted to read or write
 protected memory. This is often an indication that other memory is corrupt.
   at mainCRTStartup()

I ran the Percolator command on the attached make-pin.pin file, but the error did not arise. It is presumably specific to Windows.

[make-pin.txt.gz](https://github.com/crux-toolkit/crux-toolkit/files/902397/make-pin.txt.gz)

**Comment by kaipot** [comment]
I tried running this under Windows and the problem does occur. Standalone percolator does not throw this exception.
Crux's PercolatorAdapter inherits from Percolator's Caller; we call PercolatorAdapter::run(), and the exception happens in there. However when I copy the exact same function from Caller::run() to PercolatorAdapter, the exception doesn't occur, which seems very weird to me.

I also tried running valgrind to look for memory errors and there were a few that it found, but fixing them didn't resolve this issue. So, still looking into this.
